packages/theme/src/createTheme.mts

```diff
@@ -0,0 +1,14 @@
+import merge from 'ts-deepmerge'
+
+import { defaultTheme } from './constants.mjs'
+import type { Theme } from './types.mjs'
+
+type DeepPartial<T> = T extends object
```

Comment:
We could also use a [library](https://github.com/sindresorhus/type-fest) for that type

Reply:
Let's add this to the discussions and merge as is, to discuss this lib. I'm going to look at it. WDYT ?

Reply:
In the meantime, since it's kind of a generic type that could be used everywhere, maybe put it inside a `global.d.ts`?

Reply:
We could, I need to check if the type is embedded in the build if we do that.